x-pack/platform/plugins/shared/alerting/server/alerts_client/alerts_client.ts

```diff
@@ -684,17 +707,37 @@ export class AlertsClient<
     const isLifecycleAlert = this.ruleType.autoRecoverAlerts ?? false;
     const maxAlertLimit = this.legacyAlertsClient.getMaxAlertLimit();
 
-    const query = getMaintenanceWindowAlertsQuery({
+    const searches = getMaintenanceWindowAlertsQuery({
       executionUuid,
       ruleId,
       maintenanceWindows,
       action: isLifecycleAlert ? 'open' : undefined,
       maxAlertLimit,
     });
 
-    const response = await this.search<ScopedQueryAggregationResult>(query);
+    const responses = await this.msearch(searches);
+    const alertsByMaintenanceWindowIds: ScopedQueryAlerts = {};
+
+    responses.forEach((response) => {
+      if ('error' in response) {
+        this.options.logger.error(
+          `Error fetching scoped query alerts for maintenance windows ${this.ruleInfoMessage}: ${response.error.reason}`,
+          this.logTags
+        );
+        return;
+      }
+      response.hits.hits.forEach(({ fields }) => {
+        const mwIdField = fields![RUNTIME_MAINTENANCE_WINDOW_ID_FIELD];
```

Comment:
nit: should we use the optional accessor instead? Using `!` will throw an error and cause the whole thing to fail if we can't access the field

Reply:
I used`!` because we add this field in order to return the related mw. it should be impossible not to access it. 
But i will modify it and skip the hit if the field doesn't exist.

For the functional test: Yes we already have a functional test. It didn't fail before because we set the maxAlerts limit to 20 whereas it was 1000 for the real runtime. Now I set it to 110 in order to be sure that we exceed the fixed limit of 100 in ES.